### PR TITLE
home work 15

### PR DESCRIPTION
**Pull request by centurion84**

### CODE REVIEW

File: src/main/java/homeWork15/ProductsInCart.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    protected final String itemFieldPath = "//div[@class='float-cart__shelf-container']//div[@class='shelf-item']";
+    protected final String emptyCartViewPath = "//div[@class='float-cart__shelf-container']//p[@class='shelf-empty']";
+    protected final String priceCartPath = itemFieldPath + "[%s]//div[@class='shelf-item__price']//p";
+    protected final String titleCartPath = itemFieldPath + "[%s]//div[@class='shelf-item__details']//p[@class='title']";
+    protected final String quantityCartPath = itemFieldPath + "[%s]//div[@class='shelf-item__details']//p[@class='desc']";
+    protected final String cartViewPath = "//div[@class='float-cart float-cart--open']//div[@class='float-cart__content']";
+    protected final String removeItemButtonPath = "//div[@class='float-cart__shelf-container']//div[@class='shelf-item__del']";
+    protected final String totalCartPrice = "//p[@class='sub-price__val']";
+    protected final String closeCartPath = "//div[@class='float-cart float-cart--open']//div[@class='float-cart__close-btn']";
+    protected final String checkoutButtonPath = "//div[contains(@class,'buy-btn') and text()='Checkout']";
```
Comment (Nikots) [inline src/main/java/homeWork15/ProductsInCart.java]: Can be simplified to `//div[@class='buy-btn']` with the same effect.
Also possible to make findElement(By.classname("buy-btn"));

Also other xpath locators can be simplified in similar way.

File: src/main/java/homeWork15/ProductsInCart.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    protected final String totalPriceRegex = "\\$\\s*(\\d+)\\.\\d{2}";
+    protected final String quantityCartRegex = "\\bQuantity: (\\d+)\\b";
+
+    public static WebElement findElement(String xpath) {
+        return driver.findElement(By.xpath(xpath));
+    }
+
+    public static List<WebElement> findElements(String xpath) {
+        return driver.findElements(By.xpath(xpath));
+    }
```
Comment (Nikots) [inline src/main/java/homeWork15/ProductsInCart.java]: No need in current methods as they act exactly the same way as WebDriver methods

File: src/main/java/homeWork15/ProductsInCart.java (hunk 1 above)
Comment (centurion84) [reply src/main/java/homeWork15/ProductsInCart.java]: Shortened xpathes where it was possible and replaced some of them by classes names

File: src/main/java/homeWork15/ProductsInCart.java (hunk 2 above)
Comment (centurion84) [reply src/main/java/homeWork15/ProductsInCart.java]: Removed

**Pull request merged**